### PR TITLE
check if user is connected to the right network

### PR DESCRIPTION
**Pull request by Le-Caignec**

### CODE REVIEW

File: src/main.tsx
Comment (Le-Caignec) [inline src/main.tsx:1]: we need be agree on our prettier config

**Comment by PierreJeanjacquot** [comment]
remove init logic from LoginGuard, and use a centralized module to watch network/account changes

File: src/main.tsx
Comment (PierreJeanjacquot) [reply src/main.tsx:1]: added to backlog

Comment (Le-Caignec) [reply src/main.tsx:1]: done

File: src/features/navigation/Navigation.tsx [hunk 1]
```diff
@@ -46,12 +52,10 @@ export default function Navigation() {
...
       }}
       login={{
-        isLoggedIn: !!address && isAccountConnected,
+        isLoggedIn: !!isConnected && isAccountConnected,
         address,
-        onLoginClick: () => {
-          open();
-        },
-        onLogoutClick: () => disconnect(),
+        onLoginClick: () => open(),
```
Comment (PierreJeanjacquot) [inline src/features/navigation/Navigation.tsx]: set defult chain 134

File: src/features/loginGuard/LoginGuard.tsx [hunk 2]
```diff
@@ -1,34 +1,54 @@
...
-import { FC, ReactNode, useEffect } from 'react';
-import { useAccount } from 'wagmi';
-import { initDataProtector, selectAppIsConnected } from '../../app/appSlice';
-import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { FC, ReactNode } from 'react';
+import { useAccount, useNetwork, useSwitchNetwork } from 'wagmi';
+import { selectAppIsConnected } from '../../app/appSlice';
+import { useAppSelector } from '../../app/hooks';
 import { Box } from '@mui/material';
+import { Button } from '@iexec/react-ui-kit';
```
[the remark this answers is not in the document]
Comment (Le-Caignec) [reply src/features/loginGuard/LoginGuard.tsx]: Commit: 7ac12f04e1cb4e44b9ac10a68c74f95985bc0b30

File: src/features/navigation/Navigation.tsx (hunk 1 above)
Comment (Le-Caignec) [reply src/features/navigation/Navigation.tsx]: Commit: c50d35a682f37a4d4a1c7a232a1cefbee5021e3c

**Review by PierreJeanjacquot — APPROVED** [review]
LGTM

**Pull request merged**